### ISSUE TITLE
Add event support for "Vehicle" UnitId

### ISSUE THREAD

**Issue by Kolmarwow** [WeakAuras/WeakAuras2#4692]
**Is your feature request related to a problem? Please describe.**
When I use for example Player/Unit Info -> Health and set Unit to Specific Unit -> Vehicle, the trigger doesn't correctly turn on/off when I enter/leave vehicles presumably because of missing events like `UNIT_ENTERED_VEHICLE` / `UNIT_EXITED_VEHICLE`.

**Describe the solution you'd like**
Either the ability to select "Vehicle" under Unit or event support for Vehicle when it's written in "Specific Unit".

**Describe alternatives you've considered**
I can use the "Pet" UnitId instead which vehicles often use, but I've ran into a few edge cases where a vehicle is only using the "Vehicle" UnitId.
I can also use custom triggers instead but I'd rather not do that as I'd lose the convenient text replacements such as %health.

**Additional context**
Game Version: Wrath of the Lich King Classic
WA Version: 5.8.3

**Comment by mrbuds** [comment]
That sound reasonable, could you cite examples where this will be useful? Could be handy to know where to test the change

**Comment by Kolmarwow** [comment]
I went ingame to do more thorough testing to give examples and I realised this only happens when you're a passenger in a vehicle  without vehicle controls for you to use (such as Demolishers/Siege Engines in Wintergrasp/Isle of Conquest or multi-seated player mounts like Mekgineer's Chopper/Traveler's Tundra Mammoth).

I thought I had also seen this when I was controlling vehicles but seems I was wrong as I can't replicate that. That's not really useful then so I'll close this request until I find more useful examples of this occurring.
